### PR TITLE
Review and edit User Guide.

### PR DESCRIPTION
**Pull request by dwelsch-esi**
Edit all User Guide pages. Reorganize User Guide to guide users in applying quantization processes.

### CODE REVIEW

File: Docs/userguide/debugging_guidelines.rst [hunk 1]
```diff
@@ -4,35 +4,30 @@
 Quantization debugging guidelines
```
Comment (dwelsch-esi) [inline Docs/userguide/debugging_guidelines.rst:4]: Why is this still called "debugging"? Can we use a more accurate term, like "further quantization" or "secondary optimization"? It might help to explicitly define the goal here, something like: "To improve model accuracy to acceptable levels after application of standard quantization techniques."

At the very least I'd recommend calling it "troubleshooting", which is a less casual, more widely used terminology and is a more general term, which is appropriate here.

File: Docs/userguide/debugging_guidelines.rst [hunk 2]
```diff
@@ -4,35 +4,30 @@
...
-issues, but they are not algorithmic.
+AIMET supports various neural network quantization techniques. If the model's performance is not satisfactory after applying an AIMET quantization feature, we recommend a set of diagnostic steps to identify the bottlenecks and improve performance. These debugging steps can point to why a quantized model underperforms and help to address the underlying issues, but they are not algorithmic.
 
 Debugging workflow
 ==================
 
-The steps are shown as a flow chart in the following figure and are described in more detail below:
+The steps are shown as a flow chart in the following figure and are described in more detail below.
 
 .. image:: ../images/debugging_guidelines_1.png
```
Comment (dwelsch-esi) [inline Docs/userguide/debugging_guidelines.rst:14]: Some notes on the image:
- This would make more efficient use of page space (at least on a laptop screen) if it were arranged horizontally instead of vertically. Ideally the text in the graphic should be big enough to read by default; right now I'm having to enlarge the picture to read the text.
- The math symbols in the flowchart should be explained in the text, or better, eliminated in favor of step titles that match the text headings ("Quantizing weights vs activations", and so on). (Some abbreviation might be necessary.)
- Terms in the flowchart should correspond to identifiable operations. Specifically, "fix" is a vague description that's used several times. Numbering the steps would help with this if they follow a definite sequence.
- The boxes in the flowchart should correspond to steps in the procedure. Currently it's not clear which boxes represent which steps in the description.

File: Docs/userguide/debugging_guidelines.rst [hunk 3]
```diff
@@ -4,35 +4,30 @@
...
-   :height: 500
+   :height: 800
 
-
-1. FP32 confidence checks
-------------------------
+1. Confidence-checking FP32
```
Comment (dwelsch-esi) [inline Docs/userguide/debugging_guidelines.rst:17]: "Sanity check" is a being phased out as a non-inclusive term in the industry: https://inclusivenaming.org/
This should be changed in graphics as well.

Comment (dwelsch-esi) [reply Docs/userguide/debugging_guidelines.rst:17]: There's a new item in the backlog to link steps in the user guide to techniques in the feature guide. We should do that here as well.

File: Docs/userguide/on_target_inference.rst [hunk 4]
```diff
@@ -6,13 +6,13 @@
 On-target inference
 ###################
 
-In order to run AIMET quantized model on a target device, you need following two things:
+Running an AIMET quantized model on a target device requires two things:
 
-- an exported model,
```
Comment (dwelsch-esi) [inline Docs/userguide/on_target_inference.rst:11]: Is the exported model a single file or a collection of files? What are the file extensions?

File: Docs/userguide/on_target_inference.rst [hunk 5]
```diff
@@ -6,13 +6,13 @@
...
-- an encodings JSON file containing quantization parameters (like **encoding min/max/scale/offset**) associated with each quantizers.
+- An exported model
```
Comment (dwelsch-esi) [inline Docs/userguide/on_target_inference.rst:11]: Is the exported model a single file or a collection of files?

File: Docs/userguide/on_target_inference.rst [hunk 6]
```diff
@@ -27,140 +27,145 @@ to generate both the items. The exported model type will differ based on the fra
...
 
-|qai_hub|_ simplifies the AI model deployment on a device with runtimes like |qnn|_, |tflite|_ and |ort|_.
+|qai_hub|_ is an online platform for developers that simplifies AI model deployment on devices with runtimes like |qnn|_ (QAIRT), |tflite|_ and |ort|_.
 
-Once the AIMET exported model and an encodings JSON file have been obtained, the artifacts can be passed to the |qai_hub| for compilation,
-profiling and inference.
+Once you have obtained an AIMET exported model and a JSON encodings file, you can pass them to the |qai_hub| for compilation, profiling, and inference.
 
-Follow these instructions to `compile AIMET quantized model <https://app.aihub.qualcomm.com/docs/hub/compile_examples.html#compiling-models-quantized-with-aimet-to-tflite-or-qnn>`_ and then submit an inference job using selected device.
+Follow the `instructions <https://app.aihub.qualcomm.com/docs/hub/compile_examples.html#compiling-models-quantized-with-aimet-to-tflite-or-qnn>`_ at the Qualcomm\ |reg| AI hub to compile a model and submit an inference job using the selected device.
```
Comment (dwelsch-esi) [inline Docs/userguide/on_target_inference.rst:39]: It's not clear to me if this simulates the device or deploys it to your device. What exactly happens here?

File: Docs/userguide/on_target_inference.rst [hunk 7]
```diff
@@ -27,140 +27,145 @@ to generate both the items. The exported model type will differ based on the fra
...
-~~~~~~~~~
+See the |qnn_docs|_ for optional |qnn| HTP back-end-specific arguments.
 
-|qnn| SDK ``qnn-net-run`` tool executes the model (represented as serialized context binary) on the desired target.
 
-.. code-block:: shell
+4. Executing the model
+----------------------
+
+The |qnn| SDK ``qnn-net-run`` tool executes the model (represented as serialized context binary) on the specified target.
```
Comment (dwelsch-esi) [inline Docs/userguide/on_target_inference.rst:147]: Is running the model on a target the same as deploying it to that target?

File: Docs/userguide/quantization_workflow.rst [hunk 8]
```diff
@@ -89,70 +39,68 @@ Supported precisions to run inference on target runtimes like |qnn|_ are:
...
+2. Selecting on-target metrics
+------------------------------
 
-The figure below illustrates the recommended quantization workflow and the steps required
-to deploy the quantized model on the target device.
+Once you have achieved acceptable off-target quantized accuracy using QuantSim, consider the following
+on-target metrics. Decide which are important to your application:
+
+- Latency
+- Memory size
```
Comment (dwelsch-esi) [inline Docs/userguide/quantization_workflow.rst]: Are these the only on-target metrics that might be of interest?

File: Docs/userguide/quantization_workflow.rst [hunk 9]
```diff
@@ -89,70 +39,68 @@ Supported precisions to run inference on target runtimes like |qnn|_ are:
...
+5. Applying PTQ or QAT
+----------------------
 
-If any of the metrics are not acceptable with higher precision, begin with weights at
-INT8 precision and activations at INT16 precision. In this step, before creating the QuantSim,
-ensure that the FP32 model adheres to model specific guidelines. For instance, in PyTorch,
-QuantSim can only quantize math operations performed by :class:`torch.nn.Module` objects, while
-:class:`torch.nn.functional` calls will be incorrectly ignored. Please refer to framework specific
-pages to know more about such model guidelines.
+If any of the metrics are not acceptable with higher precision, begin with weights at INT8 precision and activations at INT16 precision.
```
Comment (dwelsch-esi) [inline Docs/userguide/quantization_workflow.rst]: We say "begin with" here, but didn't we actually begin with W16A16 in the previous step? Some more explanation might be a good idea here.

File: Docs/userguide/quantization_workflow.rst [hunk 10]
```diff
@@ -89,70 +39,68 @@ Supported precisions to run inference on target runtimes like |qnn|_ are:
...
-ensure that the FP32 model adheres to model specific guidelines. For instance, in PyTorch,
-QuantSim can only quantize math operations performed by :class:`torch.nn.Module` objects, while
-:class:`torch.nn.functional` calls will be incorrectly ignored. Please refer to framework specific
-pages to know more about such model guidelines.
+If any of the metrics are not acceptable with higher precision, begin with weights at INT8 precision and activations at INT16 precision. 
 
-If the off-target quantized accuracy metric is not meeting expectations, you can use PTQ or QAT
-techniques to improve the quantized accuracy for the desired precision. The decision between
-PTQ and QAT should be based on the quantized accuracy and runtime needs.
+If the off-target quantized accuracy metric does not meet expectations, use PTQ or QAT techniques to improve the quantized accuracy for the implemented precision. The decision to use PTQ or QAT should be based on your quantized accuracy and runtime needs.
```
Comment (dwelsch-esi) [inline Docs/userguide/quantization_workflow.rst]: Can we provide more guidance on "the decision to use PTQ or QAT"? We allude to it elsewhere, but we don't explain what accuracy and runtime criteria would dictate any particular technique.

File: Docs/userguide/quantization_workflow.rst [hunk 11]
```diff
@@ -89,70 +39,68 @@ Supported precisions to run inference on target runtimes like |qnn|_ are:
...
+-------------------------
 
-Deploy
-------
+Once the off-target quantized accuracy is satisfactory, proceed to :ref:`evaluate the
+on-target metrics<opt-guide-on-target-inference>` at this precision. If the on-target metrics still do not meet your requirements, consider further reducing the precision (for example to W8A8 or W4A8) and repeat the application of PTQ or QAT to optimize the model.
 
-Once the quantized accuracy and runtime requirements are achieved at the desired precision,
-the optimized model is ready for deployment on the target runtimes.
+Once the quantized accuracy and runtime requirements are achieved at the desired precision, deploy the optimized model on the target runtimes.
```
Comment (dwelsch-esi) [inline Docs/userguide/quantization_workflow.rst:86]: Can we provide explicit instructions for how to deploy the model, or a link to those instructions?

File: Docs/userguide/on_target_inference.rst [hunk 12]
```diff
@@ -27,140 +27,145 @@ to generate both the items. The exported model type will differ based on the fra
...
-This step generates a DLC (``*.dlc``) file that represents the model as a series of QAIRT API calls.
+This step generates a DLC file that represents the model as a series of QAIRT API calls.
 
-Please refer the |qnn_docs|_ for more details.
+See the |qnn_docs|_ for more details.
 
 
-Quantization
-~~~~~~~~~~~~
+2. Quantizating the model
```
Comment (quic-hitameht) [inline Docs/userguide/on_target_inference.rst:80]: Typo here: Quantizating

File: Docs/userguide/on_target_inference.rst [hunk 13]
```diff
@@ -27,140 +27,145 @@ to generate both the items. The exported model type will differ based on the fra
...
 
 
-Compilation
-~~~~~~~~~~~
+3. Compiling the model
+----------------------
 
-|qnn| SDK ``qnn-context-binary-generator`` tool compiles the quantized DLC (``*.dlc``) from the previous step into QNN
-serialized context binary applicable to the |qnn| HTP backend.
+The |qnn| SDK ``qnn-context-binary-generator`` tool compiles the quantized DLC (``*.dlc``) from the previous step into a QNN serialized context binary compatible with the |qnn| Hexagon tensor processor (HTP) back end.
```
Comment (quic-hitameht) [inline Docs/userguide/on_target_inference.rst:109]: Should be `backend`

File: Docs/userguide/quantization_workflow.rst [hunk 14]
```diff
@@ -89,70 +39,48 @@ Supported precisions to run inference on target runtimes like |qnn|_ are:
...
-Before using quantized integer format, it's important to ensure that the FP32 model
-and the quantized model (QuantSim object) perform similarly during the forward pass, especially
-when custom quantizers are included in the model.
+Quantize the model weights and activations to 16-bit integer (W16A16). Do the following:
 
-Set the bit-width to 16 bits for both weights and activations when creating the QuantSim.
-Then, obtain the off-target quantized accuracy metric for the quantized model and verify if
-it aligns with the FP32 model. If it doesn't, please report an issue to |aimet|_.
+1. Ensure that the FP32 model adheres to model-specific guidelines. For instance, in PyTorch QuantSim can only quantize math operations performed by :class:`torch.nn.Module` objects, while :class:`torch.nn.functional` calls will be incorrectly ignored. See framework-specific pages to learn more about such model guidelines.
+2. Once the model conforms to guidelines, create a quantization simulation (QuantSim) version of your model with the bit-width set to 16 bits for both weights and activations (W16A16). See :ref:`<quantsim-workflow>`.
```
Comment (dwelsch-esi) [inline Docs/userguide/quantization_workflow.rst:67]: I don't think there are pages that describe how to to the QuantSim steps the same way there are in the Feature Guide. Can we write standalone pages using that template for:
- Creating a QuantSim model, and
- Testing an off-target model's accuracy?

File: Docs/userguide/quantization_workflow.rst [hunk 15]
```diff
@@ -89,70 +39,48 @@ Supported precisions to run inference on target runtimes like |qnn|_ are:
...
+3. Ensure that the original FP32 model and the quantized model (QuantSim object) perform similarly during the forward pass, especially when custom quantizers are included in the model.
```
Comment (dwelsch-esi) [inline Docs/userguide/quantization_workflow.rst:68]: How does the user do this? Do we provide instructions somewhere?
